app/main.py

```diff
@@ -2,0 +1,36 @@
+class Animal:
+    alive = []  # all alive animals
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> None:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def reduce_health(self, amount: int) -> None:
+        self.health -= amount
+        if self.health <= 0:
+            self.health = 0
+            self.die()
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, prey: str) -> None:
```

Comment:
The `prey` parameter is typed as `str`, but it should be an instance of `Herbivore`. Change the type annotation to `Herbivore`.